jisantuc/panrec
# Publish docker container on successful master builds

Overview
-----

This PR publishes a docker container for running `panrec` on successful builds of `master`.

Notes
-----

For some reason docker doesn't pass the arguments unless you specify the entrypoint, even if i put _the same entrypoint_ in the dockerfile. It's not inconvenient enough to figure out why.

First test isn't filtering to master, since I want to see if it actually works

Testing
-----

- build the dockerfile -- `docker build -t panrec ./`
- somewhere else, e.g., in the Raster Foundry repo, run the docker container to do a conversion:

```fish
docker run --entrypoint ./panrec -it \
    -v (pwd)/app-backend/datamodel/src/main/scala:/opt/data/in \
    -v (pwd)/app-backend/datamodel/target/scala-2.12:/opt/data/out \
    --entrypoint './panrec'
    panrec \
    -i scala -o ts-interface /opt/data/in /opt/data/out
```

- observe that it at least attempted to do what you said

Closes #20
